File: test/four/app/src/main/java/com/pan3d/material/DynamicBaseTexItem.java
package com.pan3d.material;

import com.pan3d.scene.Scene3D;

public class DynamicBaseTexItem {

    public String paramName;
    public TexItem target;
    public TextureRes textureRes;
    public Scene3D scene3D;
    public DynamicBaseTexItem(Scene3D val){
        scene3D=val;
    }

}

File: new/src/Pan3d/mateial/DynamicBaseTexItem.ts
module Pan3d {
    export class DynamicBaseTexItem {
        public target: TexItem;
        public paramName: string;
        public textureRes: TextureRes;

      
        public scene3D:Scene3D;
        public constructor(value:Scene3D){

            this.scene3D=value;
        }


        public get texture(): WebGLTexture {
            if (this.textureRes) {
                return this.textureRes.texture;
            }
            return null;
        }

    }
}